pkg/network/pasta/pasta.go

```diff
@@ -19,6 +20,35 @@ import (
 	"github.com/rootless-containers/rootlesskit/v2/pkg/network/iputils"
 )
 
+type Features struct {
+	// Has `--host-lo-to-ns-lo` (introduced in passt 2024_10_30.ee7d0b6)
+	// https://passt.top/passt/commit/?id=b4dace8f462b346ae2135af1f8d681a99a849a5f
+	HasHostLoToNsLo bool
+}
+
+func DetectFeatures(binary string) (*Features, error) {
+	if binary == "" {
+		return nil, errors.New("got empty pasta binary")
+	}
+	realBinary, err := exec.LookPath(binary)
+	if err != nil {
+		return nil, fmt.Errorf("pasta binary %q is not installed: %w", binary, err)
+	}
+	cmd := exec.Command(realBinary, "--help")
+	b, err := cmd.CombinedOutput()
+	s := string(b)
+	if err != nil {
+		return nil, fmt.Errorf(
+			"command \"%s --help\" failed, make sure pasta is installed: %q: %w",
+			realBinary, s, err,
+		)
+	}
+	f := Features{
+		HasHostLoToNsLo: strings.Contains(s, "--host-lo-to-ns-lo"),
```

Comment:
If it makes things simpler, you can also test the whole thing with `pasta --host-lo-to-ns-lo --version`, that is:

```
$ pasta --host-lo-to-ns-lo --version >/dev/null 2>&1 && echo option is there
option is there
$ pasta --invalid-option --version >/dev/null 2>&1 || echo option is not there
option is not there
```

By the way, sorry for breaking this. I didn't really expect users/test to actually bind *to 127.0.0.1* in the container. :(

Reply:
Thanks, updated